db/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/db/InntektsmeldingRepository.kt

```diff
@@ -32,99 +34,134 @@ class InntektsmeldingRepository(
             .labelNames("method")
             .register()
 
-    fun lagreInntektsmelding(
+    fun hentNyesteInntektsmelding(forespoerselId: UUID): Inntektsmelding? =
+        requestLatency.recordTime(InntektsmeldingRepository::hentNyesteInntektsmelding) {
+            transaction(db) {
+                hentNyesteImQuery(forespoerselId)
+                    .firstOrNull()
+                    ?.getOrNull(InntektsmeldingEntitet.dokument)
+            }
+        }
+
+    fun hentNyesteEksternEllerInternInntektsmelding(forespoerselId: UUID): Pair<Inntektsmelding?, EksternInntektsmelding?> =
+        requestLatency.recordTime(InntektsmeldingRepository::hentNyesteEksternEllerInternInntektsmelding) {
+            transaction(db) {
+                InntektsmeldingEntitet
+                    .select(InntektsmeldingEntitet.dokument, InntektsmeldingEntitet.eksternInntektsmelding)
+                    .where { InntektsmeldingEntitet.forespoerselId eq forespoerselId.toString() }
+                    .orderBy(InntektsmeldingEntitet.innsendt, SortOrder.DESC)
+                    .limit(1)
+                    .map {
+                        Pair(
+                            it[InntektsmeldingEntitet.dokument],
+                            it[InntektsmeldingEntitet.eksternInntektsmelding],
+                        )
+                    }.firstOrNull()
+            }.orDefault(Pair(null, null))
+        }
+
+    fun oppdaterJournalpostId(
+        innsendingId: Long,
+        journalpostId: String,
+    ) {
+        requestLatency.recordTime(InntektsmeldingRepository::oppdaterJournalpostId) {
+            val antallOppdatert =
+                transaction(db) {
+                    InntektsmeldingEntitet.update(
+                        where = { InntektsmeldingEntitet.id eq innsendingId },
+                    ) {
+                        it[InntektsmeldingEntitet.journalpostId] = journalpostId
+                    }
+                }
+
+            if (antallOppdatert == 1) {
+                "Lagret journalpost-ID '$journalpostId' i database.".also {
+                    logger.info(it)
+                    sikkerLogger.info(it)
+                }
+            } else {
+                "Oppdaterte uventet antall ($antallOppdatert) rader med journalpost-ID '$journalpostId'.".also {
+                    logger.error(it)
+                    sikkerLogger.error(it)
+                }
+            }
+        }
+    }
+
+    fun lagreEksternInntektsmelding(
```

Comment:
Diffen i denne fila er mye vanskeligere å lese enn den hadde behøvd å være synes jeg. 

Essensen er at `lagreInntektsmelding` er fjernet og at jeg har lagt til fire nye metoder nederst i fila: For å lagre og hente skjema, for å hente nyeste innsendingId for en forespørsel og for å oppdatere med beriket dokument.